packages/schemas/src/SourceScheduleSchema.ts

```diff
@@ -8,7 +8,27 @@ export const sourceScheduleSchema = z.object({
   /**
    * Startup delay for the source
    */
-  initialDelayMs: z.number()
+  initialDelayMs: z.number(),
+  /**
+   * Enable retries
+   * @default true
+   */
+  retryEnabled: z.boolean().optional().default(true),
+  /**
+   * interval between retries.  This will rise exponentially on every failure.
+   * @default 5 mins
+   */
+  retryDelayMins: z.number().optional().default(5),
+  /**
+   *  Maximum number of retry attempts
+   * @default 100
+   */
+  maxRetries: z.number().optional().default(100),
+  /**
+   * If true, after a successful retry the retry logic is reset to its initial settings
```

Comment:
NIT comment was a little unclear until I looked up the RXJS api

I think this means the retry counter is reset to zero once it recovers.

Reply:
What would the use-case where you would not resetOnSuccess ?

Reply:
Updated the comment.  I think we would always want to reset it to true, but we have a variety of sources so wanted to be as flexible as possible.